### ISSUE TITLE
When inserting approach when flying the AP decides sometimes to go in random direction

### ISSUE THREAD

**Issue by uzytkownik** [flybywiresim/a32nx#4802]
### Aircraft Version

Development

### Build info

```json
{
  "built": "2021-05-12T20:09:03+00:00",
  "ref": "master",
  "sha": "3c615945e45603cca52a3f3190ff33651d414976",
  "actor": "sidnov",
  "event_name": "manual"
}
```

### Describe the bug

During inserting the approach into F-Plan while flying AP decides to go in random heading. It is still in managed mode and I can see the correct F-Plan on navigation display.

Switching mode of AP does not fix the issue - only reengaging the AP.

### Expected behavior

AP flies the F-Plan

### Steps to reproduce

I don't know. I reproduced the issue a few times but I could not do it consistently.

**Comment by aguther** [comment]
Can you please provide a bit more info. What did you see on the FMA of the PFD in those situations? What happened when you selected the other approach? How did the FMA then change?

See also the following page about reporting issues:
https://docs.flybywiresim.com/start/autopilot-fbw/

**Comment by uzytkownik** [comment]
@aguther  The link is dead.

I will check next time it reproes.

**Comment by aguther** [comment]
There seem to be some issues with the website right now. For me it still works…

**Comment by CP241** [comment]
Hi there,
I seem to have a similar problem. I currently use Dev version d2e2c84 and EDDH airport from Aerosoft. The input flight plan is as follows:  Depart RW23 on SID HAM1B to the HAM VOR, then a left turn to DH457 for an ILS15. Altitude selected 4000ft. Managed mode. Prior reaching HAM VOR the aircraft veers left to DH457. I the intervene by selecting heading mode and vector aircraft to final RW15. I descend to 3000ft. Wen on an intercept heading of 180 I select APP. aircraft now wildly climbs back to 4000ft then descends again. Speed oscillates wildly, too. Aircraft becomes unmanagable. I must say I get quite frustrated. This has been going on for over a month and new Dev releases provide no improvement. Maybe someone can fly the approach and see for him-herself.

**Comment by aguther** [comment]
> Hi there,
> I seem to have a similar problem. I currently use Dev version [d2e2c84](https://github.com/flybywiresim/a32nx/commit/d2e2c84fa3e9f39563dfaafeb8b5c83b126ea3ec) and EDDH airport from Aerosoft. The input flight plan is as follows: Depart RW23 on SID HAM1B to the HAM VOR, then a left turn to DH457 for an ILS15. Altitude selected 4000ft. Managed mode. Prior reaching HAM VOR the aircraft veers left to DH457. I the intervene by selecting heading mode and vector aircraft to final RW15. I descend to 3000ft. Wen on an intercept heading of 180 I select APP. aircraft now wildly climbs back to 4000ft then descends again. Speed oscillates wildly, too. Aircraft becomes unmanagable. I must say I get quite frustrated. This has been going on for over a month and new Dev releases provide no improvement. Maybe someone can fly the approach and see for him-herself.

I'm getting frustrated as well bc it takes *a lot* of time to find out what is going on because often not much info is provided beside complaints. It's quite hard to tell what is going on without more info. The dev version does not improve because I never experienced what you're saying...

Can you please provide what your FMA was saying over those maneuvers?

Please refer to the following document and ensure requirements are fulfilled:
https://docs.flybywiresim.com/start/autopilot-fbw/

**Comment by CP241** [comment]
Hi,

The problem of the aircraft not following what has been programmed is an old one, and, honestly, is very annoying. Don’t get me wrong, I think the guys are doing quite a good job, considering we are reaping the benefits for free. But, alas, emphasis must be on getting things right without annoyance to the users. This problem of the aircraft not following what has been programmed is an old one, and has yet to be solved. Also, this must be known to the programmers, and I would have thought a solution would have been found some time ago. We get updates every day, and I appreciate there being bugs in the version, but this I find, is a major issue that has been ongoing for too long. The aircraft just does not follow the programmed route, climbs/descends willy-nilly, speed often not constant, major issues that must be known. Further, there is no indication why all this happens, and manual intervention often does not help. I have been flying the approach I mentioned numerous times, and there have been numerous new versions of the dev. version, but the problem still persists. I sometimes wonder, if those who do all the programming, actually fly the aircraft. I am not happy at all, sorry to say. Thanks for your attention.

Claus Pohl 

 �

**Comment by Saschl** [comment]
It is being worked on. The whole navigation system is very complex though. The issues with the flight plan especially need a complete reimplementation as we can not really modify the default behaviour in this case. That being said I do fly the plane everyday and do encounter these issues occasionally but honestly most of the time It’s dependent on the flight plan and another one works just fine then.

**Comment by aguther** [comment]
@CP241 Thank you for you feedback.

I personally invested around 500 hours of my free time into the custom fly-by-wire and custom autopilot and auto-thrust system and I think it's a quite good result already comparing it even with other commercial add-ons (at least for normal law). I did a lot of flying and I estimate I did 200-300 times an Autoland. Now this is just myself, there are other developers doing also a very very high number of hours in development for flight model, engines, sounds, support.

Now where you have a point is the LNAV and VNAV, this is related to flight management and the custom Autopilot needs to be fed with that. You can imagine that I'm also not happy about it, because it makes the custom autopilot not look good.

That's the reason why there is now working ongoing to do this, but also understand that I have more in my life than this project. So this is now done by some other developers for the time being. Many of us did not program for FSX or P3D so you can imagine there is beside the results a deep learning curve that has to be taken.

People like you are not helpful for motivation by the way. I understand that you're not happy and this should be solved and I hope you can imagine that it's not that easy to solve. You're by the way kindly invited to help.

You're statements have been confusing by the way, as I said the one part is guidance, but if you use the custom autopilot this has no influence on the vertical mode. Again I take a look, but this is only possible if we get those information.

If it's just about complaining and throwing around bad mood, please stop it. There is a simple solution for this: use a different plane.

**Comment by CP241** [comment]
Hi again,

Thanks for your swift response. My observation/complaint has not been directed at you personally. I and many others appreciate the ongoing work and the time spent by the developers. By the same token, it is highly annoying to still find the same problem after so many updates. Maybe the priorities should focus more on getting the flight module (LNAV/VNAV) correct rather than getting eye-candy like the flypad, for example.
I also find it rather time consuming to exactly describe a problem, which actually can best be viewed/experienced by just simply flying it (as is my case). It would take about 20 minutes to fly the approach as explained and one would see first-hand the problem. Documenting all the wrong moves on the part of the aircraft, including pictures, would take much longer, and you would still have to sift thru all that and, possibly, make sense of it. I am not a programmer myself, so no succour from this side. 
Finally, I but hope you take my comments in the right spirit, and not personally. As mentioned above, I and many others value your free contribution.
Have a great weekend.

Claus Pohl



Sent from myMail for iOS


Saturday, 15. May 2021, 15:51 +0200 from ***@***.***  ***@***.***>:

**Comment by Benjozork** [comment]
> Maybe the priorities should focus more on getting the flight module (LNAV/VNAV) correct rather than getting eye-candy like the flypad, for example.

Those are worked on by completely different people. This isn't how that works. We can't put everyone one a task they do not have the skills of knowledge for. It'll slow us down even more.

As the person working on the LNAV system, it's honestly annoying to see those comments. The system is being re-made from the ground up and requires a lot of things to be adapted or even sometimes remade entirely. I have a full time job and an internship to finish. I have to work on this entire system by myself with a project license I don't like and the very difficult job of keeping performance acceptable, with little to no help from Microsoft whatsoever.

It would be more productive, I think, to inquire about what work exactly is going on, instead of assuming that we do not know what we're doing and that we're putting the priorities in the wrong place.

The FSLabs A320 took 6 years to make originally. This has been going on for 9 months now. It's not going to be perfect for a while and it's not like we can see "plane does X thing wrong" and a perfect solution comes right up in our minds... This is all tangled up in engineering secrets and complex documentation that contradicts itself.

**Comment by ferryvanaesch** [comment]
For balance, I find the opposite. I learned to double check the flight plan, and how to deal with the inaccuracies that the MSFS flight plan manager throws at you. Changing things around in the flight plan for a waypoint that you're quite close to is a big no for me as that often results in a wide turn to nowhereland; I suspect it wants to do a 360 to intercept the waypoint I was too close to, but I never let it run to find out. When I have to, and I often have to when flying an approach on vatsim, I just switch off the autopilot before changing the flight plan. If I see that the fp manager is happy with what I did I reenable the autopilot. If it isn't, I continue flying manually, and I patiently wait for the huge task of a new fp manager to be implemented. 

I truly appreciate all the work that the volunteers throw at this for absolutely zero in return, and I see a lot of improvements  almost 7 days per week. 

The nice thing about an open source project is that instead of complaining, you can actively do something to help out. If you don't have the coding skills, then detailed and constructive feedback helps just as well.

**Comment by looking1977** [comment]
Hi!
I just made a video today flying on VATSIM from LOWW to BIKF. After I got information from controller to expect ASRUN2K arrival for rwy19 at BIKF I have started to setup the MCDU. After inserting the arrival the plane starts to turn left, however the immediate waypoint has not changed. It looks the system wants to continue from previous waypoint.
You can find the video on this link, hope it helps: https://youtu.be/sFBNyxiEsu0

**Comment by looking1977** [comment]
Hello!
On the way back from BIKF to LOWW the same happened, you can find it captured on video: https://youtu.be/jv863R8-DLw
There are multiple strange behaviours can be identified:
1) After Selecting arrival, the plane again starts to turn
2) Starts to turn even before adding it to the route (pushing the INSERT)
3) After inserting only the active leg remains solid green, the rest showed with dashed lines
4) on direct page can't select the next immediate waypoint, need to be entered manually

I made a second vid where already flying on the transition: https://youtu.be/hqxuoL7lRuU
1) In NAV mode the route still shows with dashed line
2) Right ND does not show the route, only up to MABOD point.
3) strange behaviour of AP, not "really" following the route

In all video the latest development built was flown available at the time of recording.

Hope guys the above videos are helping to understand the issue a bit more.

Thank you!

**Comment by airfirefighter** [comment]
same problem here every time i take off without selecting landing runway and selecting on route the plane stop following  the route and goes back to the last waypoint also  i loose all the other waypoints.....

**Comment by looking1977** [comment]
just wondering, the submitted video was sufficient, anyone could reproduce the issue?
anything else I can do or provide? :)

**Comment by tracernz** [comment]
> just wondering, the submitted video was sufficient, anyone could reproduce the issue?
> anything else I can do or provide? :)

It's fine. It's a known issue that will be fixed by our own flight plan/guidance manager currently under development. Unfortunately we have no control over sequencing with the in-sim flight plan manager so it can't be fixed until that work is complete.

**Comment by marcsoler** [comment]
Not sure if related, but this happens many times when I change the destination approach or runway. I can easily reproduce this:

https://user-images.githubusercontent.com/1502124/126900666-4c9dc753-fafb-48a9-a6a6-fb0caa7dc627.mp4

But maybe I'm just doing something wrong

**Comment by NUTTTTTTTT** [comment]
@marcsoler Hiya, Tracernz comment from June 7th would apply in this case too which is below for you. This is just one of the Asobo flight plan manager issues unfortunately.  

> It's a known issue that will be fixed by our own flight plan/guidance manager currently under development. Unfortunately we have no control over sequencing with the in-sim flight plan manager so it can't be fixed until that work is complete.

**Comment by marcsoler** [comment]
@NUTTTTTTTT yes I was suspecting that since the same also happens on other aircrafts. Thanks for the heads-up

**Comment by maxpain** [comment]
Same problem. Any workarounds? I can't fly on VATSIM/IVAO because I can't select STAR and approach before a flight.

**Comment by FRYYYYYY** [comment]
Same here... no known workaround available :-(

**Comment by tracernz** [comment]
See https://github.com/flybywiresim/a32nx/issues/4802#issuecomment-856153435

**Comment by 2hwk** [comment]
Fixed by #4953 closing ☑️
